Should obtain SA token from token secret

This tool does not seem to work on Openshift, as it attempts to obtain the SA token from the Openshift registry image pull secret. This is an automated secret created for all SA in Openshift clusters. I think the algorithm to find the correct secret (containing a token) must be improved.

````
$ proxy-kubeconfig-generator --kubeconfig-secret-key kubeconfig --namespace erikbo --server 'https://capsule-proxy.capsule-system.svc:9001' --server-tls-secret-namespace capsule-system --server-tls-secret-name capsule-proxy --serviceaccount gitops-reconciler
panic: secret gitops-reconciler-dockercfg-qv5nr does not contain a token

goroutine 1 [running]:
main.main()
        /home/erikbo/go/pkg/mod/github.com/maxgio92/proxy-kubeconfig-generator@v0.1.0/main.go:68 +0x434
````

Ref. https://capsule.clastix.io/docs/guides/flux2-capsule/#the-recipe

--- a/pkg/utils/utils.go
+++ b/pkg/utils/utils.go
@@ -47,31 +47,14 @@ func BuildClientConfig() (*rest.Config, error) {
 }
 
 func GetServiceAccountTokenSecret(clientSet *kubernetes.Clientset, serviceAccountName string, namespace string) (*corev1.Secret, error) {
-	serviceAccount, err := clientSet.CoreV1().ServiceAccounts(namespace).Get(
-		context.Background(),
-		serviceAccountName,
-		metav1.GetOptions{},
-	)
+	secretList, err := clientSet.CoreV1().Secrets(namespace).List(context.Background(), metav1.ListOptions{})
 	if err != nil {
 		return nil, err
 	}
 
-	if len(serviceAccount.Secrets) < 1 {
-		return nil, fmt.Errorf("no secret found for the service account %s in namepsace %s", serviceAccount.Name, serviceAccount.Namespace)
-	}
-
-	for _, secret := range serviceAccount.Secrets {
-		saSecret, err := clientSet.CoreV1().Secrets(namespace).Get(
-			context.Background(),
-			secret.Name,
-			metav1.GetOptions{},
-		)
-		if err != nil {
-			continue
-		}
-
-		if saSecret.Type == corev1.SecretTypeServiceAccountToken {
-			return saSecret, nil
+	for _, secret := range secretList.Items {
+		if secret.Type == corev1.SecretTypeServiceAccountToken && secret.Annotations[corev1.ServiceAccountNameKey] == serviceAccountName {
+			return &secret, nil
 		}
 	}
 

